models/bucket.py

```diff
@@ -100,7 +110,22 @@ def check_deadline(self):
         based on the attribute frequency.
 
         '''
-        pass
+        current_datetime = datetime.now()
+        if (self.frequency is not None) and (self.deadline < current_datetime):
```

Comment:
I would do this with a dictionary instead of doing so many if statements, it's faster